### PR TITLE
uniswap tokens by volume

### PR DESCRIPTION
**Pull request by skylarbarrera**
this pr fixes issues regarding search results and liquidity info(sorting low & high). 

1.  we now don't sort by liquidity if we dont have any liquidity info, so in this case we only get Verified and Unverified sections(before we were defaulting to low liquidity).

2. I  changed the univ2 tokens query to pull in by USDVolume, this fixes the lag where Sushi & SuperRare would be in low liquidity until we got their liq info. Also should result in more relevant results showing up before the shit coins.

3. I removed results where the derivedETH value = 0, we were already doing this for totalLiquidity so this makes sense and filters out more 0 val tokens.

example for sushi & yearn search: https://cloud.skylarbarrera.com/Screen-Recording-2021-09-10-16-27-28.mp4

fixes RNBW-1605

### CODE REVIEW

File: src/handlers/uniswap.ts [hunk 1]
```diff
@@ -412,19 +412,20 @@ export const getAllTokens = async () => {
   const { dispatch } = store;
   try {
     let dataEnd = false;
-    let lastId = '';
+    let lastUSDVolume = '1000000000000';
```
Comment (mikedemarais) [inline src/handlers/uniswap.ts:415]: @skylarbarrera how did we come up with this magic number? can you leave a comment above it explaining what this number is

File: src/redux/uniswap.js [hunk 2]
```diff
@@ -57,7 +57,7 @@ const parseTokens = tokens => {
   tokens.forEach(token => {
     const tokenAddress = toLower(token.id);
     const metadata = getTokenMetadata(tokenAddress);
-    if (token.totalLiquidity === '0') return;
+    if (token.totalLiquidity === '0' || token.derivedETH === '0') return;
```
Comment (mikedemarais) [inline src/redux/uniswap.js:60]: noice

**Review by mikedemarais — CHANGES_REQUESTED** [review]
this is very dank ser and i appreciate you doing this very very much ser

this pull request looks exquisite 😗👌 

all i request is that you add a comment explaining the `let lastUSDVolume = '1000000000000';` magic number,,, and then it is good to merge fren

File: src/handlers/uniswap.ts (hunk 1 above)
Comment (skylarbarrera) [reply src/handlers/uniswap.ts:415]: i just set an extremely high number so we're getting all the data, currently the highest USD volume is in the 10's millions so 1 trill should be more than enough. I'll add a comment 😘

File: src/handlers/uniswap.ts [hunk 3]
```diff
@@ -412,19 +412,21 @@ export const getAllTokens = async () => {
...
     while (!dataEnd) {
       let result = await uniswapClient.query({
         query: UNISWAP_ALL_TOKENS,
         variables: {
           first: UniswapPageSize,
-          lastId,
+          lastUSDVolume,
         },
       });
+      //console.log('result: ', result);
```
Comment (brunobar79) [inline src/handlers/uniswap.ts:426]: WTF!

File: src/apollo/queries.js [hunk 4]
```diff
@@ -204,14 +204,20 @@ export const UNISWAP_PRICES_QUERY = gql`
...
 
 export const UNISWAP_ALL_TOKENS = gql`
-  query tokens($first: Int!, $lastId: String!) {
-    tokens(first: $first, where: { id_gt: $lastId }) {
+  query tokens($first: Int!, $lastUSDVolume: String!) {
+    tokens(
+      first: $first
+      orderBy: tradeVolumeUSD
+      orderDirection: desc
+      where: { tradeVolumeUSD_lt: $lastUSDVolume }
```
Comment (brunobar79) [inline src/apollo/queries.js:212]: _lte  instead of _lt

**Review by brunobar79 — APPROVED** [review]
🤝

File: src/handlers/uniswap.ts (hunk 3 above)
Comment (skylarbarrera) [reply src/handlers/uniswap.ts:426]: 😅

**Pull request merged**